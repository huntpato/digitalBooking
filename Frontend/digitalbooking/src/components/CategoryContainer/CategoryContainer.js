import { CategoryCard } from "../CategoryCard/CategoryCard";
import useFetch from "use-http";
import styles from "./CategoryContainer.module.css";
import { Link } from "react-router-dom"

export const CategoryContainer = () => {
  const options = { headers:{
    "X-Requested-With": "XMLHttpRequest"
  }};
  const url = "https://cors-everywhere-me.herokuapp.com/g4bookingapp-env-2.eba-4mbmxg4f.us-east-1.elasticbeanstalk.com/category";
  const { loading, data = [] } = useFetch(url, options, []);
  const { searchCategory, searchByType, categoryCards } = styles;

  return (
    <div className={searchCategory}>
      <section className={searchByType}>
        <h4>Buscar por tipo de alojamiento</h4>
        {loading && <p>Cargando Categor√≠as...</p>}
        <div className={categoryCards}>
          { data && data.map((category, index) => (
            <Link key= {index} to= {`/category/${category.id}`} style={{textDecoration: 'none', color:'black'}} ><CategoryCard
              key={index}
              image={category.photoUrl}
              alt={category.description}
              title={category.name}
              description={category.countProducts}
            /></Link>
          ))}
        </div>
      </section>
    </div>
  );
};
