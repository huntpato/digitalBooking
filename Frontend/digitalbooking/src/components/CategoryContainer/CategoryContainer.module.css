.searchCategory {
  display: flex;
  flex-direction: row;
  height: fit-content;
  justify-content: center;
}

.searchByType {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 40px 0px 40px 0px;
  width: 90vw;
}

.categoryCards {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: repeat(2, 1fr);
  grid-column-gap: 15px;
  grid-row-gap: 15px;
  width: 90vw;
}

.searchByType h4 {
  margin-bottom: 15px;
  align-self:flex-start;
}


@media (max-width: 768px) {

  .categoryCards {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    grid-template-rows: repeat(4, 1fr);
    grid-column-gap: 15px;
    grid-row-gap: 15px;
    width: 90vw;
  }

  .searchByType {
    padding: 10px 0px 30px 0px;
    width: 90vw;
  }

}

@media (max-width: 576px) {

  .categoryCards {
    display: grid;
    grid-template-columns: repeat(1, 1fr);
    grid-template-rows: repeat(1, 1fr);
    grid-column-gap: 15px;
    grid-row-gap: 25px;
    width: 90vw;
  }

  .searchByType {
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding: 10px 0px 30px 0px;
    margin: 20px 0px 40px 0px;
  }

  .searchByType h4 {
    margin-left: 0%;
  }

  .searchCategory {
    flex-direction: column;
    flex-wrap: wrap;
    justify-content: space-around;
    align-items: center;
  }
}

@media (max-width: 480px) {

  .categoryCards {
    display: grid;
    grid-template-columns: repeat(1, 1fr);
    grid-template-rows: repeat(1, 1fr);
    grid-column-gap: 15px;
    grid-row-gap: 25px;
    width: 90vw;
  }

  .searchByType {
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding: 10px 0px 30px 0px;
    margin: 20px 0px 40px 0px;
  }

  .searchByType h4 {
    margin-left: 0%;
  }

  .searchCategory {
    flex-direction: column;
    flex-wrap: wrap;
    justify-content: space-around;
    align-items: center;
  }
}
