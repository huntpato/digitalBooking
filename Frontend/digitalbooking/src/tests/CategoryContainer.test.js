import React from 'react';
import { shallow } from 'enzyme';
import { CategoryContainer } from '../components/CategoryContainer/CategoryContainer'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals';


jest.mock("react-router-dom", () => ({
    useLocation: jest.fn().mockReturnValue({
      pathname: "/",
      search: "",
      hash: "",
      state: null,
      key: "5nvxpbdafa",
    }),
  }));
  
describe('Pruebas en componente <CategoryContainer />', () => {
    const wrapper= shallow(<CategoryContainer/>);
   
    test('Comprobar que <CategoryContainer /> renderiza correctamente.', () => {

        expect( wrapper ).toMatchSnapshot();
    })

    test('Verificar que el componente carga la clase de CSS correctamente', () => {

        const div = wrapper.find('div').at(0);
        expect(div.hasClass('searchCategory')).toBe(true);
    })
    
})
