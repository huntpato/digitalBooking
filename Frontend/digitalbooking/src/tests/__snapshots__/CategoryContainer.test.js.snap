// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <CategoryContainer /> Comprobar que <CategoryContainer /> renderiza correctamente. 1`] = `
<div
  className="searchCategory"
>
  <section
    className="searchByType"
  >
    <h4>
      Buscar por tipo de alojamiento
    </h4>
    <p>
      Cargando Categor√≠as...
    </p>
    <div
      className="categoryCards"
    />
  </section>
</div>
`;
